test/MusicStore.Test/StoreControllerTest.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            // Arrange
+            var genreName = "Genre 1";
+
+            var dbContext = _serviceProvider.GetRequiredService<MusicStoreContext>();
+            CreateTestGenres(numberOfGenres: 3, numberOfAlbums: 3, dbContext: dbContext);
+
+            var controller = new StoreController()
+            {
+                DbContext = dbContext,
+            };
+
+            // Act
+            var result = await controller.Browse(genreName);
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.ViewName);
+
+            Assert.NotNull(viewResult.ViewData);
+            var viewModel = Assert.IsType<Genre>(viewResult.ViewData.Model);
+            Assert.Equal(genreName, viewModel.Name);
+            Assert.NotNull(viewModel.Albums);
+            Assert.Equal(3, viewModel.Albums.Count);
+        }
+
+        [Fact]
+        public async Task Details_ReturnsHttpNotFoundWhenNoAlbumData()
+        {
+            // Arrange
+            var albumId = 0;
+            var controller = new StoreController()
+            {
+                DbContext = _serviceProvider.GetRequiredService<MusicStoreContext>(),
+                Cache = _serviceProvider.GetRequiredService<IMemoryCache>(),
+            };
+
+            // Act
+            var result = await controller.Details(albumId);
+
+            // Assert
+            Assert.IsType<HttpNotFoundResult>(result);
+        }
+
+        [Fact]
+        public async Task Details_ReturnsAlbumDetail()
+        {
+            // Arrange
+            var albumId = 1;
+
+            var dbContext = _serviceProvider.GetRequiredService<MusicStoreContext>();
+            var genres = CreateTestGenres(numberOfGenres: 3, numberOfAlbums: 3, dbContext: dbContext);
+
+            var cache = _serviceProvider.GetRequiredService<IMemoryCache>();
+
+            var controller = new StoreController()
+            {
+                DbContext = dbContext,
+                Cache = cache,
+            };
+
+            // Act
+            var result = await controller.Details(albumId);
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.ViewName);
+
+            Assert.NotNull(viewResult.ViewData);
+            var viewModel = Assert.IsType<Album>(viewResult.ViewData.Model);
+            Assert.NotNull(viewModel.Genre);
+            var genre = genres.SingleOrDefault(g => g.GenreId == viewModel.GenreId);
+            Assert.NotNull(genre);
+            Assert.NotNull(genre.Albums.SingleOrDefault(a => a.AlbumId == albumId));
+            Assert.Null(viewModel.Artist);
+
+            var cachedAlbum = cache.Get<Task<Album>>("album_1");
+            Assert.NotNull(cachedAlbum);
+            Assert.Equal(albumId, cachedAlbum.Result.AlbumId);
+        }
+
+        private Genre[] CreateTestGenres(int numberOfGenres, int numberOfAlbums, DbContext dbContext)
```

Comment:
The return value of `CreateTestGenres` seems not used any where. Can this be changed to `private void`?